File: tools/parity/dispatch/terminal-surface-id.ts
// TS dispatch for the terminal-surface-id parity module: maps the shared vector
// function names to the real `src/shared/terminal-surface-id.ts` exports so the
// harness compares the live TS reference against the Rust port. All three
// functions take a single string argument, so `input` is the raw string.

import {
  isWebTerminalSurfaceTabId,
  toHostSessionTabId,
  toWebTerminalSurfaceTabId
} from '../../../src/shared/terminal-surface-id'

export function dispatch(fn: string, input: unknown): unknown {
  switch (fn) {
    case 'toWebTerminalSurfaceTabId':
      return toWebTerminalSurfaceTabId(input as string)
    case 'toHostSessionTabId':
      return toHostSessionTabId(input as string)
    case 'isWebTerminalSurfaceTabId':
      return isWebTerminalSurfaceTabId(input as string)
    default:
      throw new Error(`unknown function ${fn}`)
  }
}

File: rust/crates/orca-dispatch/src/modules/terminal_surface_id.rs
//! Parity dispatch for `orca_core::terminal_surface_id` vs
//! `src/shared/terminal-surface-id.ts`.

use orca_core::terminal_surface_id::{
    is_web_terminal_surface_tab_id, to_host_session_tab_id, to_web_terminal_surface_tab_id,
};
use serde_json::{json, Value};

pub fn dispatch(function: &str, input: &Value) -> Value {
    // Every function takes a single string arg; vectors carry it as the raw input.
    let arg = input.as_str().unwrap_or_default();
    match function {
        "toWebTerminalSurfaceTabId" => Value::String(to_web_terminal_surface_tab_id(arg)),
        "toHostSessionTabId" => Value::String(to_host_session_tab_id(arg)),
        "isWebTerminalSurfaceTabId" => Value::Bool(is_web_terminal_surface_tab_id(arg)),
        other => json!({ "__parity_error__": format!("unknown function {other}") }),
    }
}
